[CSS Zoom] Fix zoom factor for inherited letter-spacing

vitorroriz opened the pull request:
#### a3cf1b778f1aafcc94f97fbd5ac71c8d8bdfc435
<pre>
[CSS Zoom] Fix zoom factor for inherited letter-spacing
<a href="https://rdar.apple.com/162085369">rdar://162085369</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=300278">https://bugs.webkit.org/show_bug.cgi?id=300278</a>

Reviewed by Antti Koivisto.

We want to store the unzoomed letter-spacing value at RenderStyle
and the zoomeed value at FontCascade.

Test: imported/w3c/web-platform-tests/css/css-text/parsing/letter-spacing-inherited-computed.html
* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/letter-spacing-inherited-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/letter-spacing-inherited-computed.html: Added.
* Source/WebCore/style/StyleBuilderState.cpp:
(WebCore::Style::BuilderState::updateFontForSizeChange):
* Source/WebCore/style/values/text/StyleLetterSpacing.cpp:
(WebCore::Style::CSSValueConversion&lt;LetterSpacing&gt;::operator):
* Source/WebCore/style/values/text/StyleLetterSpacing.h:

Canonical link: <a href="https://commits.webkit.org/301269@main">https://commits.webkit.org/301269@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/610490696b0d00ff73afe06adf28d59f096a3988

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125440 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45107 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35848 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132297 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77325 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6be35a72-b18a-4244-b2aa-07cd2be615a2) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127316 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45794 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53669 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95521 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63443 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/450ca84e-2f35-4061-92b3-a3c0c420c6bc) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128393 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36594 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112181 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76047 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/366ad005-d497-44ed-9cf8-f9a78397a882) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/35495 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30354 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75775 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106365 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30581 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134980 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52242 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40023 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/104000 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52681 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108397 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103749 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26419 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49112 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27412 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49401 "Hash 61049069 for PR 51906 does not build (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52140 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57924 "") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51491 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54847 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53186 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/654223ddbcafd91f0a467a5aa7d56e21b57e501d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51906

vitorroriz committed:
[CSS Zoom] Fix zoom factor for inherited letter-spacing
rdar://162085369
https://bugs.webkit.org/show_bug.cgi?id=300278

Reviewed by Antti Koivisto.

We want to store the unzoomed letter-spacing value at RenderStyle
and the zoomeed value at FontCascade.

Test: imported/w3c/web-platform-tests/css/css-text/parsing/letter-spacing-inherited-computed.html
* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/letter-spacing-inherited-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/letter-spacing-inherited-computed.html: Added.
* Source/WebCore/style/StyleBuilderState.cpp:
(WebCore::Style::BuilderState::updateFontForSizeChange):
* Source/WebCore/style/values/text/StyleLetterSpacing.cpp:
(WebCore::Style::CSSValueConversion<LetterSpacing>::operator):
* Source/WebCore/style/values/text/StyleLetterSpacing.h:

Canonical link: https://commits.webkit.org/301269@main
